### PR TITLE
Bug 1946232: VM List can consume very high CPU

### PR DESCRIPTION
**Pull request by glekner**
blocked by https://github.com/openshift/console/pull/9758

### CODE REVIEW

**Comment by yaacov** [comment]
#9758 - merged

/lgtm

**Comment by glekner** [comment]
@zcahana hey,
I'm noticing the `printableStatus` is `Provisioning` when the test is expecting it to be `Stopped`
the vm has no `runStrategy`, why is the status `Provisioning`?
image

**Comment by yaacov** [comment]
/retest

p.s.
@glekner if the `printableStatus` show `provisionining` when `spec.running=false` we can address thins discrepancy in the `conditions` column by stating "(icon) Stopped" in the conditions column

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/view/vm.ts [hunk 1]
```diff
@@ -8,11 +8,16 @@ import { wizard } from './wizard';
 
 export const waitForStatus = (status: string, vmData?: VirtualMachineData, timeout?: number) => {
   const timeOut = timeout || VM_ACTION_TIMEOUT.VM_IMPORT;
-  cy.contains(detailsTab.vmStatus, status, { timeout: timeOut }).should('exist');
   if (status === VM_STATUS.Running) {
     const { name, namespace } = vmData;
     cy.waitForLoginPrompt(name, namespace);
   }
+
+  if (status === VM_STATUS.Starting) {
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/view/vm.ts:16]: a. https://github.com/kubevirt/user-guide/pull/439/files
maybe we may report starting when real status is provisioning ?
we should look at the logic for reporting "starting" and see if in some cases we should report "provisioning" instead ?

b. can we make a new method 
`export const waitForStatuses = (statuses: string[], vmData?: VirtualMachineData, timeout?: number) => {`
and use it when we can continue if we see one of the correct statuses ?
e.g.
continue to next step if vm status is in `["starting", "provisioning"]` acceptable statuses list ...

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/vm/advanced-wizard.ts [hunk 2]
```diff
@@ -87,7 +87,7 @@ describe('Test VM creation', () => {
     });
   });
 
-  [urlVM, registryVM, pvcVM, pxeVM].forEach((vmData) => {
+  [urlVM, registryVM, pvcVM].forEach((vmData) => {
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/vm/advanced-wizard.ts:90]: NOTE:
this is skipping the PXE test, we will need to re-introduce this test once we fix the test flakynes
I don't think the PXE test flakynes is connected to this PR

**Comment by zcahana** [comment]
> I'm noticing the printableStatus is Provisioning when the test is expecting it to be Stopped
the vm has no runStrategy, why is the status Provisioning?

@glekner (cc @yaacov) sorry for the delayed response (was on vacation 😎).

This behavior is actually by design. The different VM states aren't mutually exclusive, some of them coexist (e.g., Stopped and Provisioning, or Running and Migrating). Since we include only one state in `printableStatus`, we choose the one we think is more significant. For this particular case, Provisioning was chosen over Stopped since the DataVolumes for the VM are currently being prepared (provisioned, populated with data, etc). Once this is done, the `printableStatus` field should change back to `Stopped`.

**Pull request merged**